daemon/userconsent_test.go

```diff
@@ -0,0 +1,323 @@
+	assert.True(t, cm.cfg.AutoConnectData.ID != 0)
+
+	err := acc.doLightLogout()
+
+	assert.NoError(t, err)
+	assert.True(t, cm.saved)
+	assert.Zero(t, len(cm.cfg.TokensData))
+	assert.Zero(t, cm.cfg.AutoConnectData.ID)
+}
+
+func TestPrepareDaemonIfConsentNotCompleted(t *testing.T) {
+	category.Set(t, category.Unit)
+
+	truePtr := true
+	tests := []struct {
+		name               string
+		initialConfig      config.Config
+		loadErr            error
+		apiInsights        core.InsightsAPI
+		apiErr             error
+		authLoggedIn       bool
+		expectedSaved      bool
+		expectedConsentSet bool
+		expectedConsentVal bool
+		expectedTokensLen  int
+		expectedAutoConnID int64
+	}{
+		{
+			name:               "consent already completed",
+			initialConfig:      config.Config{AnalyticsConsent: &truePtr},
+			expectedSaved:      false,
+			expectedConsentSet: true,
+			expectedConsentVal: true,
+			expectedTokensLen:  0,
+			expectedAutoConnID: 0,
+		},
+		{
+			name:               "standard country -> set consent",
+			initialConfig:      config.Config{AnalyticsConsent: nil},
+			apiInsights:        &fakeInsights{insightsResult: &core.Insights{CountryCode: "ca"}},
+			expectedSaved:      true,
+			expectedConsentSet: true,
+			expectedConsentVal: true,
+			expectedTokensLen:  0,
+			expectedAutoConnID: 0,
+		},
+		{
+			name: "GDPR country & logged in -> do light logout",
+			initialConfig: config.Config{
+				AnalyticsConsent: nil,
+				TokensData:       map[int64]config.TokenData{42: {}},
+				AutoConnectData:  config.AutoConnectData{ID: 42},
+			},
+			apiInsights:        &fakeInsights{insightsResult: &core.Insights{CountryCode: "fr"}},
+			authLoggedIn:       true,
+			expectedSaved:      true,
+			expectedConsentSet: false,
+			expectedTokensLen:  0,
+			expectedAutoConnID: 0,
+		},
+		{
+			name:               "GDPR country & not logged in",
+			initialConfig:      config.Config{AnalyticsConsent: nil},
+			apiInsights:        &fakeInsights{insightsResult: &core.Insights{CountryCode: "fr"}},
+			authLoggedIn:       false,
+			expectedSaved:      false,
+			expectedConsentSet: false,
+			expectedTokensLen:  0,
+			expectedAutoConnID: 0,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cm := &fakeConfigManager{cfg: tt.initialConfig, loadErr: tt.loadErr}
+			api := tt.apiInsights
+			authChk := &fakeAuthChecker{loggedIn: tt.authLoggedIn}
+
+			acc := &AnalyticsConsentChecker{cm: cm, insightsAPI: api, authChecker: authChk}
+			acc.PrepareDaemonIfConsentNotCompleted()
+
+			assert.Equal(t, cm.saved, tt.expectedSaved)
+			if tt.expectedConsentSet {
+				assert.NotNil(t, cm.cfg.AnalyticsConsent)
+				assert.Equal(t, *cm.cfg.AnalyticsConsent, tt.expectedConsentVal)
+			} else {
+				assert.Nil(t, cm.cfg.AnalyticsConsent)
+			}
+			assert.Equal(t, len(cm.cfg.TokensData), tt.expectedTokensLen)
+			assert.Equal(t, cm.cfg.AutoConnectData.ID, tt.expectedAutoConnID)
+		})
+	}
+}
+
+type fakeConfigManager struct {
```

Comment:
Why not to use https://github.com/NordSecurity/nordvpn-linux/blob/main/test/mock/config_manager.go?

Reply:
True, done